### PR TITLE
domain logic: final bindings

### PR DESCRIPTION
**Pull request by vsalavatov**
I've tested it by hand and it (at least search haha) works good! Of course, someday we'll write unit tests for all these things.......

### CODE REVIEW

File: cmd/httpapi/main.go [hunk 1]
```diff
@@ -25,22 +32,74 @@ func readCryptoKey(privateKeyPath string, publicKeyPath string) (privateKeyBytes
...
-    unarXivUsecases := smartUsecases.SmartUsecases{AccountStorage: memory.NewAccountsRepo(), Auth: a}
+    authUsecases := auth.NewUsecases(memory.NewAccountsRepo(), jwtAuth)
+    articleRepo := memory.NewArticleRepo()
+    historyRepo := memory.NewHistoryRepo()
+    artSubsRepo := memory.NewArticleSubscriptionRepo()
+    searchSubsRepo := memory.NewSearchSubscriptionRepo()
+    updatesRepo := repository.NewUpdatesRepoThroughQueries(articleRepo, artSubsRepo, searchSubsRepo, historyRepo)
 
-    httpApi := httpapi.New(&unarXivUsecases)
+    if articleDataPath != nil {
```
Comment (M0nsteR) [inline cmd/httpapi/main.go:55]: Oh no, please spare me reading all these nested spaghetti.. Refactor immediately! No more than 1 (1,5 max) level of nesting should be ever approved.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Flat is better than nested. (c) Python Zen

File: cmd/httpapi/main.go (hunk 1 above)
Comment (vsalavatov) [reply cmd/httpapi/main.go:55]: fixed

Comment (vsalavatov) [reply cmd/httpapi/main.go:55]: oops, wait a minute please, gotta fix merge conflicts

Comment (vsalavatov) [reply cmd/httpapi/main.go:55]: done

**Review by M0nsteR — APPROVED** [review]
Good to go.

**Pull request merged**